Code/Tools/ProjectManager/Source/ProjectExportWorker.cpp

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <ProjectExportWorker.h>
+#include <ProjectManagerDefs.h>
+#include <PythonBindingsInterface.h>
+#include <ProjectUtils.h>
+
+#include <QDir>
+#include <QFile>
+#include <QProcess>
+#include <QProcessEnvironment>
+#include <QTextStream>
+#include <QThread>
+
+namespace O3DE::ProjectManager
+{
+    ProjectExportWorker::ProjectExportWorker(const ProjectInfo& projectInfo)
+        : QObject()
+        , m_projectInfo(projectInfo)
+    {
+    }
+
+    void ProjectExportWorker::ExportProject()
+    {
+        auto result = ExportProjectForPlatform();
+
+        if (result.IsSuccess())
+        {
+            emit Done();
+        }
+        else
+        {
+            emit Done(result.GetError());
+        }
+    }
+
+    AZ::Outcome<QString, QString> ProjectExportWorker::GetLogFilePath() const
+    {
+        QDir logFilePath(m_projectInfo.m_path);
+        // Make directories if they aren't on disk
+        if (!logFilePath.cd(ProjectBuildPathPostfix))
+        {
+            bool madeDirectory = logFilePath.mkpath(ProjectBuildPathPostfix);
+            if (!madeDirectory)
+            {
+                return AZ::Failure("Unable to make log directory for project build path");
+            }
+            logFilePath.cd(ProjectBuildPathPostfix);
+        }
+        if (!logFilePath.cd(ProjectBuildPathCmakeFiles))
+        {
+            bool madeDirectory = logFilePath.mkpath(ProjectBuildPathCmakeFiles);
+            if (!madeDirectory)
+            {
+                return AZ::Failure("Unable to make log directory for cmake files path");
+            }
+            logFilePath.cd(ProjectBuildPathCmakeFiles);
+        }
+        return AZ::Success(logFilePath.filePath(ProjectExportErrorLogName));
+    }
+
+    void ProjectExportWorker::QStringToAZTracePrint([[maybe_unused]] const QString& error)
+    {
+        AZ_Trace("Project Manager", error.toStdString().c_str());
+    }
+
+    AZ::Outcome<void, QString> ProjectExportWorker::ExportProjectForPlatform()
+    {
+        AZ::Outcome<QString, QString> logFilePathQuery = GetLogFilePath();
+        if(!logFilePathQuery.IsSuccess())
+        {
+            QString errorMessage = tr("%1: %2").arg(LogPathFailureMsg).arg(logFilePathQuery.GetError());
```

Comment:
`tr` won't give you any benefit with the `"%1: %2"` format string. It will perform `tr` on `%1 %2`, return it as a QString object, and when the `args` are applied, it will not translate the args. `tr` should instead be applied on the `args`